packages/deployments/contracts/contracts/messaging/connectors/wormhole/BaseWormhole.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.17;
+
+import {IWormholeRelayer} from "../../interfaces/ambs/wormhole/IWormholeRelayer.sol";
+import {IWormholeReceiver} from "../../interfaces/ambs/wormhole/IWormholeReceiver.sol";
+
+import {GasCap} from "../GasCap.sol";
+import "forge-std/console.sol";
+
+abstract contract BaseWormhole is GasCap, IWormholeReceiver {
+  // ============ Storage ============
+  address public immutable relayerAddress;
+  uint16 public immutable MIRROR_CHAIN_ID;
+  mapping(bytes32 => bool) public processedWhMessages;
+
+  // ============ Constructor ============
+  constructor(address _relayer, uint256 _gasCap, uint16 _mirrorChainId) GasCap(_gasCap) {
+    relayerAddress = _relayer;
+    MIRROR_CHAIN_ID = _mirrorChainId;
+  }
+
+  // ============ Public Fns ============
+  /**
+   * @notice This function is called to receive messages through the wormhole relayer module
+   * https://book.wormhole.com/technical/evm/relayer.html
+   */
+  function receiveWormholeMessages(
+    bytes memory payload,
+    bytes[] memory additionalVaas,
+    bytes32 sourceAddress,
+    uint16 sourceChain,
+    bytes32 deliveryHash
+  ) public payable override {
+    require(sourceChain == MIRROR_CHAIN_ID, "!source chain");
+    require(msg.sender == relayerAddress, "!relayer");
```

Comment:
so, when we call `processMessageFromRoot` on the backend the call stack will be:

1. Call some fn on the wormhole relayer
2. That calls `_receiveWormholeMessages`
3. `_processMessageFrom` processes the aggregate or spoke root, depending on direction

is the relayer different than the AMB?

Reply:
its not, I refactored to remove `relayerAddress` to make that clear.